### PR TITLE
Batch link support

### PR DESCRIPTION
**Pull request by ericanderson**
```ts
const { data: employees } = await client(Employee).fetchPage({
  pageSize: 100,
});

logger.debug("Fetching the bulk links");

for await (
  const {
    object,
    linkApiName,
    otherObjectApiName,
    otherObjectPk,
  } of client.__UNSTABLE_getBulkLinks(employees, [
    "ventures",
    "amishsSyncGroup",
  ])
) {
  logger.info(
    `Found link ${object.$objectType}:${object.$primaryKey} <- (${linkApiName}) -> ${otherObjectApiName}:${otherObjectPk}`,
  );
}
```

### CODE REVIEW

File: packages/client/src/createClient.ts [hunk 1]
```diff
@@ -71,10 +70,11 @@ export function createClientInternal(
   }
 
   const client: Client = Object.defineProperties<Client>(
-    clientFn as UnstableClient,
+    clientFn as Client,
     {
       ctx: {
         value: clientCtx,
+        writable: false,
```
Comment (ssanjay1) [inline packages/client/src/createClient.ts:77]: Why did we need to add this?

File: packages/client/src/ontology/OntologyProvider.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ export interface OntologyProvider {
    */
   getObjectDefinition: (
     apiName: string,
-  ) => Promise<ObjectTypeDefinition<any>>;
+  ) => Promise<ObjectTypeDefinition<any> & { rid: string }>;
```
Comment (ssanjay1) [inline packages/client/src/ontology/OntologyProvider.ts:35]: Just double checking, we're only adding rid here for internal use when we want to get object defs for batch links right?

**Review by ssanjay1 — APPROVED** [review]
Approved, left some comments/questions

File: packages/client/src/__unstable/createBulkLinksAsyncIterFactory.ts [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        ontologyBranchRid: undefined,
+        owningRid: undefined,
+        reportUsage: undefined,
+        workstateRid: undefined,
+      },
+      responseOptions: {
+        includeObjectSetEntities: true,
+        includeUsageCost: false,
+      },
+      pageSize: 1000,
```
[the remark this answers is not in the document]
Comment (ericanderson) [reply packages/client/src/__unstable/createBulkLinksAsyncIterFactory.ts:91]: Im not sure. We can iterate on this part later. We need feedback on this anyway.

File: packages/client/src/createClient.ts (hunk 1 above)
Comment (ericanderson) [reply packages/client/src/createClient.ts:77]: I was trying to protect caches a bit better but I think this is superseeded by other changes I made for the omniapi anyway.

File: packages/client/src/ontology/OntologyProvider.ts (hunk 2 above)
Comment (ericanderson) [reply packages/client/src/ontology/OntologyProvider.ts:35]: Correct. Thats why I didn't add it to the actual `ObjectTypeDefinition`, especially as thats in the api package and shared with `legacy-client`. We should not need this in an ideal world.

**Review by ericanderson — COMMENTED** [review]
Notes

**Pull request merged**